### PR TITLE
Adding bufr2ioda test output to r2d2 database in soca test

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Adds existing bufr2ioda test output in soca test to r2d2 database

Files changed:

test/soca/CMakeLists.txt
test/soca/create_obsdb.py

Fulfills implicit requirement of https://github.com/NOAA-EMC/godas/issues/412

Comments on approach:

New dictionaries are created separately from the existing one to process into the database. This is in part to have a simple way to deal with the new ioda files being located in a different subdirectory from the existing input,  and that two of the new sources (bufr_tesacprof and bufr_trkobprof) have a slightly different filename format and are from a different date. This is done with the understanding that the the database consists of the files output into build/test/soca/obs/r2d2-shared/gdasapp/ob/soca/P1D/ and there is no other hidden output or state that is clobbered by calling ufsda.r2d2.store multiple times.

The obs_types are presumed to be ok with the format of bufr_\<source\>, which is different from the existing ones. Also, as discussed with Guillaume, the date format in the new sources are in unix epoch seconds, vs. the ISO-whatever form of existing ones.

### CODE REVIEW

File: test/soca/create_obsdb.py [hunk 1]
```diff
@@ -37,5 +37,34 @@
...
+                         'source_dir': '../../testoutput/',
+                         'source_file_fmt': '{source_dir}/{obs_type}_{year}{month}{day}.nc',
+                         'type': 'ob',
+                         'database': 'shared',
+                         'provider': 'gdasapp',
+                         'experiment': 'soca',
+                         'obs_types': ['bufr_dbuoyprof',
+                                       'bufr_mbuoybprof',
+                                       'bufr_sfcships',
+                                       'bufr_sfcshipsu']})
```
Comment (ShastriPaturi) [inline test/soca/create_obsdb.py]: @AndrewEichmann-NOAA, the bufr_tesacprof and bufr_trkobprof are missing.
You mentioned about dates and filename different. Can you elaborate on what you meant?

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

File: test/soca/create_obsdb.py [hunk 2]
```diff
@@ -37,5 +37,34 @@
                                        'salt_profile_fnmoc',
                                        'icec_EMC',
                                        'icefb_GDR']})
+    ufsda.r2d2.store(obsstore)
 
+# Create the test R2D2 database for output from bufr2ioda tests
```
Comment (guillaumevernieres) [inline test/soca/create_obsdb.py:42]: You don't need to redefine the entire dictionary, it's almost the same as the one above bar a few keys.
Maybe something like this instead:
```python
obsstore['source_file_fmt'] = '{source_dir}/{obs_type}_{year}{month}{day}.nc'
obstore['obs_types'] = ...
```

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
@AndrewEichmann-NOAA : A few minor comments from myself and @ShastriPaturi before we can merge your work.

File: test/soca/create_obsdb.py (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply test/soca/create_obsdb.py]: bufr_tesacprof and bufr_trkobprof are handled separately, in the section below this. The ioda files here have the form` {obs_type}_{year}{month}{day}.nc`, like bufr_dbuoyprof_20180415.nc, but the other two are `{obs_type}_{year}{month}.nc` (missing the day), and as such have different entries for `source_file_fmt`. Hence the two calls to `ufsda.r2d2.store()`.

File: test/soca/create_obsdb.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply test/soca/create_obsdb.py:42]: done

**Comment by guillaumevernieres** [comment]
@AndrewEichmann-NOAA , if you don't click on the circular arrow in front of the reviewers' name, we won't know your PR is ready for an other round of reviews ...

review

If you're done, re-request our reviews ...

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres Sorry, I thought you were automatically notified of updates

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres Sorry, I thought you were automatically notified of updates

@AndrewEichmann-NOAA , I am. But if you don't click on that arrow, I'll assume it's not ready to re-review.

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Comment by guillaumevernieres** [comment]
@ShastriPaturi , are you going to review again or can I just merge this PR?

**Review by ShastriPaturi — APPROVED** [review]
Did not check the dates in bufr_<tesac,trkob>prof data.
I approve though.

**Comment by ShastriPaturi** [comment]
A quick comment, the ioda converted bufr data in r2d2 DB will have "Location" instead of "nlocs" as the dimension.

**Comment by AndrewEichmann-NOAA** [comment]
@ShastriPaturi Ok, i'll make another PR out of that

**Pull request merged**